engine/src/main/java/org/terasology/rendering/dag/gsoc/PropertyChangeEvent.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.gsoc;
+
+import org.terasology.module.sandbox.API;
+
+@API
+public class PropertyChangeEvent extends java.beans.PropertyChangeEvent {
```

Comment:
I am sorry, but what's the point of this class?

Reply:
This class probably isn't needed, but I still haven't tested the events for module based class. There's this problem when using 3rd party library during runtime from within a module...forgot I could maybe whitelist it? I'll take a look

Reply:
Yes, by default you can't use 3rd party libraries from modules. However, you can whitelist them explicitly. Way cleaner.